Indexing/TermSubstitutionTree.hpp

```diff
@@ -32,9 +31,12 @@ namespace Indexing {
  * Note that unlike LiteralSubstitutionTree, TermSubstitutionTree does
  * not (yet) carry out sort checking when attempting to find unifiers, generalisations
  * or instances. In particular, if the query or result is a variable, it is the callers'
- * responsibility to ensure that the sorts are unifiable/matchable
+ * responsibility to ensure that the sorts are unifiable/matchable 
+ * (edit: if the caller inserts a TypedTermList instead of a TermList, this will be handled automatically.)
```

Comment:
Should we ever insert an untyped `TermList`, then?

Reply:
Joe says that this is legal, and that the "expected thing" happens. He may consider changing the interface at some later point to ensure that a sort is always inserted explicitly.